libraries/testing/tester.cpp

```diff
@@ -1304,7 +1304,8 @@ namespace eosio::testing {
          auto [privkey, pubkey, pop] = get_bls_key(name);
          local_finalizer_keys[pubkey.to_string()] = privkey.to_string();
       }
-      control->set_node_finalizer_keys(local_finalizer_keys);
+      if (control)
+         control->set_node_finalizer_keys(local_finalizer_keys);
```

Comment:
`control` is not set when the node is `closed`. By checking control here we don't have to worry whether the node is closed and use `set_node_finalizers` instead of accessing the member directly [here](https://github.com/AntelopeIO/spring/blob/1d19e44ec9a10e454a7c8316db2aeade2369ad29/unittests/savanna_finalizer_policy_tests.cpp#L145-L150) for example.

Reply:
`set_node_finalizers` does set the member variable `_node_finalizers`, which is applied to the node when it starts (when `open()` is called) [here](https://github.com/AntelopeIO/spring/blob/1d19e44ec9a10e454a7c8316db2aeade2369ad29/unittests/savanna_cluster.cpp#L48). On that call `control` will be set.

Reply:
I think `savanna_cluster` should check `control` and not call `tester::set_node_finalizers(node_finalizers)`

Reply:
I don't mind changing it, but I feel we might as well do the check in `tester` since it is the one accessing `control`, rather than in callers of `tester::set_node_finalizers` of which there could possibly be more than one.

Reply:
Then it should shortcut out or have an `if` around the entire implementation as the rest of the method does nothing.

Reply:
Oh, I see what you mean now. Will shortcut.